### PR TITLE
debugger

### PR DESCRIPTION
**Pull request by gdementen**
resurrect PR I accidentally closed before merging.

I wonder if the removal of "from larray import *" should be mentioned in the changelog. I guess our current users will not notice since
they do "from larray import *" in their code anyway but... What do you think?

### CODE REVIEW

File: larray_editor/api.py [hunk 1]
```diff
@@ -11,8 +11,9 @@
 import larray as la
 
 from larray_editor.editor import REOPEN_LAST_FILE, MappingEditor, ArrayEditor
+from larray_editor.traceback_tools import extract_stack, extract_tb, StackSummary
 
-__all__ = ['view', 'edit', 'compare', 'REOPEN_LAST_FILE', 'run_editor_on_exception']
+__all__ = ['view', 'edit', 'debug', 'compare', 'REOPEN_LAST_FILE', 'run_editor_on_exception']
```
Comment (alixdamman) [inline larray_editor/api.py]: Don't forget to update the larray project (`viewer/__init__.py` + `api.rst`)

**Comment by alixdamman** [comment]
What would happen if a user types ``view(<session object>)`` ? Is the larray library loaded with the session object?

**Comment by gdementen** [comment]
> What would happen if a user types `view(<session object>)` ? Is the larray library loaded with the session object?

No. In that case, the top-level larray functions are not loaded automatically. I don't know whether users use that idiom though.

I could change that in that case if you think it's worth it...

**Comment by alixdamman** [comment]
> I don't know whether users use that idiom though.

I do but it's maybe only me.

> I could change that in that case if you think it's worth it...

You removed that automatic call to ``from larray import *`` for performance reason I guess. 
Let's try and see if users complain (except me ;) )

**Comment by gdementen** [comment]
> You removed that automatic call to `from larray import *` for performance reason I guess.

No, it has nothing to do with performance (at least currently because larray is a dependency of the editor and is loaded anyway before we hit the console). It is just that it is more correct that way for view()/debug()/edit(). For example, when using any of those functions in the middle of code which does **not** use "from larray import *" (or which do not use larray at all for that matter), this is annoying because it: 

1. overrides user variables/functions (or functions from other libs loaded with from x import y) with larray functions causing weird and very hard to diagnose problems. The most usual case I get bitten by is that I have a variable x which is overridden by larray's deprecated "axis reference factory".

2. makes dir() or top-level functions tab completion very long and annoying even if the code being inspected/debugged was done cleanly with few functions.

> Let's try and see if users complain (except me ;) )

No, I think that's more convenient to still do it for Session objects. I just overlooked that case. Thanks for mentioning this. I will remove that commit from the PR and open an issue to properly handle that later.

**Pull request merged**